pyo3-macros-backend/src/from_pyobject.rs

```diff
@@ -243,7 +243,7 @@ impl<'a> Container<'a> {
         for i in 0..len {
             let error_msg = format!("failed to extract field {}.{}", quote!(#self_ty), i);
             fields.push(quote!(
-                s.get_item(#i).extract().map_err(|inner| {
+                s.get_item(#i)?.extract().map_err(|inner| {
```

Comment:
I think this probably needs a `.and_then` or something rather than returning early, because we still need the error message:

```suggestion
                s.get_item(#i).and_then(PyAny::extract).map_err(|inner| {
```